soroban-sdk/src/testutils.rs

```diff
@@ -416,3 +419,98 @@ pub trait Deployer {
+    /// Returns the flags for account_id.
+    pub fn flags(&self) -> u32 {
+        self.env
+            .host()
+            .with_mut_storage(|storage| {
+                let k = Rc::new(xdr::LedgerKey::Account(xdr::LedgerKeyAccount {
+                    account_id: self.account_id.clone(),
+                }));
+
+                let entry = storage.get(
+                    &k,
+                    soroban_env_host::budget::AsBudget::as_budget(self.env.host()),
+                )?;
+
+                match entry.data {
+                    xdr::LedgerEntryData::Account(ref e) => Ok(e.flags.clone()),
+                    _ => panic!("expected account entry but got {:?}", entry.data),
+                }
+            })
+            .unwrap()
+    }
+
+    /// Adds the flag specified to the existing issuer flags
+    pub fn set_flag(&self, flag: IssuerAccountFlags) {
+        self.overwrite_issuer_flags(self.flags() | (flag as u32))
+    }
+
+    /// Clears the flag specified from the existing issuer flags
+    pub fn clear_flag(&self, flag: IssuerAccountFlags) {
+        self.overwrite_issuer_flags(self.flags() & (!(flag as u32)))
+    }
+
+    pub fn address(&self) -> crate::Address {
+        xdr::ScAddress::Account(self.account_id.clone())
+            .try_into_val(&self.env.clone())
+            .unwrap()
+    }
+
+    /// Sets the issuer flags field.
+    /// Each flag is a bit with values corresponding to [xdr::AccountFlags]
+    ///
+    /// Use this to test interactions between trustlines/balances and the issuer flags.
+    fn overwrite_issuer_flags(&self, flags: u32) {
+        if u64::from(flags) > xdr::MASK_ACCOUNT_FLAGS_V17 {
+            panic!(
+                "issuer flags value must be at most {}",
+                xdr::MASK_ACCOUNT_FLAGS_V17
+            );
+        }
+
+        self.env
+            .host()
+            .with_mut_storage(|storage| {
+                let k = Rc::new(xdr::LedgerKey::Account(xdr::LedgerKeyAccount {
+                    account_id: self.account_id.clone(),
+                }));
+
+                let mut entry = storage
+                    .get(
+                        &k,
+                        soroban_env_host::budget::AsBudget::as_budget(self.env.host()),
+                    )?
+                    .as_ref()
+                    .clone();
+
+                match entry.data {
+                    xdr::LedgerEntryData::Account(ref mut e) => e.flags = flags,
+                    _ => panic!("expected account entry but got {:?}", entry.data),
+                }
+
+                storage.put(
+                    &k,
+                    &Rc::new(entry),
+                    None,
+                    soroban_env_host::budget::AsBudget::as_budget(self.env.host()),
+                )?;
+                Ok(())
+            })
+            .unwrap();
+    }
+}
+
+pub struct StellarAssetContract {
+    pub address: crate::Address,
+    pub issuer: StellarAssetIssuer,
```

Comment:
Same here, can we make these private for same reasons above, any change becomes breaking, etc. We can make them `pub(crate)`.